### PR TITLE
Enable always true assert checks in Travis

### PR DESCRIPTION
**Pull request by JackStouffer**

### CODE REVIEW

**Comment by JackStouffer** [comment]
@Hackerpilot Looks like a bug in dscanner. It's still looking at things in `unittest`s

File: std/algorithm/sorting.d [hunk 1]
```diff
@@ -880,7 +880,7 @@ template multiSort(less...) //if (less.length > 1)
         }
 
         static if (funs.length == 0)
-            assert("No sorting predicate provided");
+            assert(false, "No sorting predicate provided");
```
Comment (Hackerpilot) [inline std/algorithm/sorting.d]: Another reason that implicit conversion to `bool` is bad language design.

Comment (Hackerpilot) [inline std/algorithm/sorting.d]: And shouldn't this be a `static assert` since less is a template argument?

**Comment by JackStouffer** [comment]
@Hackerpilot Fixed

File: .dscanner.ini [hunk 2]
```diff
@@ -63,6 +63,6 @@ auto_ref_assignment_check="disabled" ; FIXME
 ; Checks for incorrect infinite range definitions
 incorrect_infinite_range_check="enabled"
 ; Checks for asserts that are always true
-useless_assert_check="disabled"; FIXME
+useless_assert_check="skip-unittest"
```
Comment (wilzbach) [inline .dscanner.ini]: Do you plan to set it to `enabled` in the next stage? :)

Comment (JackStouffer) [inline .dscanner.ini]: Not really. The two remaining ones are demonstrations in ddoc unit tests.

Comment (Hackerpilot) [inline .dscanner.ini]: "skip-unittest" makes sense for this check because asserts are often used to check that something compiles.

**Comment by Hackerpilot** [comment]
Blocked because of a D-Scanner bug.

**Comment by wilzbach** [comment]
@JackStouffer would you be so kind to rebase this? After @Hackerpilot deployed the fix so quickly :)

**Comment by JackStouffer** [comment]
@Hackerpilot @wilzbach Done. Ready to merge.

**Pull request merged**